suite-native/module-onboarding/src/screens/OnboardingXPub.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Pressable, View, Alert } from 'react-native';
+
+import { useFocusEffect } from '@react-navigation/native';
+
+import { Screen, StackProps } from '@suite-native/navigation';
+import { Box, Chip, Input, InputWrapper, Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { CryptoIcon } from '@trezor/icons';
+import { NetworkKey } from '@suite-common/wallet-config';
+
+import { Camera } from '../components/Camera';
+import { OnboardingStackParamList, OnboardingStackRoutes } from '../navigation/routes';
+
+const xpubContainerStyle = prepareNativeStyle(utils => ({
+    padding: utils.spacings.medium,
+}));
+
+const coinStyle = prepareNativeStyle(_ => ({
+    flexDirection: 'row',
+}));
+
+const cameraStyle = prepareNativeStyle(_ => ({
+    marginTop: 20,
+    marginBottom: 45,
+}));
+
+const cameraPlaceholderStyle = prepareNativeStyle(utils => ({
+    height: 329,
+    borderRadius: utils.borders.radii.medium,
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: utils.colors.gray800,
+}));
+
+const chipStyle = prepareNativeStyle(utils => ({
+    flex: 1,
+    borderRadius: utils.borders.radii.small,
+}));
+
+const DEFAULT_XPUB_INPUT_TEXT = '';
+const DEFAULT_CURRENCY_SYMBOL = 'btc';
+const DEFAULT_XPUB_RESULT = null;
+
+export const OnboardingXPub = ({
+    navigation,
+}: StackProps<OnboardingStackParamList, OnboardingStackRoutes.OnboardingXPub>) => {
+    const [xpubResult, setXpubResult] = useState<string | null>(DEFAULT_XPUB_RESULT);
+    const [selectedCurrencySymbol, setSelectedCurrencySymbol] =
+        useState<NetworkKey>(DEFAULT_CURRENCY_SYMBOL);
+    const [inputText, setInputText] = useState<string>(DEFAULT_XPUB_INPUT_TEXT);
+    const [cameraRequested, setCameraRequested] = useState<boolean>(false);
+    const { applyStyle } = useNativeStyles();
+
+    useFocusEffect(
+        useCallback(() => {
+            const setDefaultValues = () => {
```

Comment:
It is necessary to set default at least for the camera because it is not showing then anymore when you go back in a navigator - the camera is not showing even when it is allowed. So that is the reason why I am setting `setCameraRequested(false)` - to rerender QR code scan (camera) to be requested again. But I think the better solution is to set it when focus is lost so I will change it to:

```
    useFocusEffect(
        useCallback(() => {
            const setDefaultValues = () => {
                setSelectedCurrencySymbol(DEFAULT_CURRENCY_SYMBOL);
                setInputText(DEFAULT_XPUB_INPUT_TEXT);
                setCameraRequested(false);
            };

            return () => setDefaultValues();
        }, []),
    );
```